### DISCUSSION TITLE
Get updated/changed data in functions

### DISCUSSION THREAD

**Discussion by FabiKel** [appwrite/appwrite#5615] [Q&A] [answered]
Hi, is there an option to get the attribute of an document that was changed on the update event? 
The function event data only gives the new DocumentObject but I need to now which attributes were updated to do specific tasks.

**Comment by byawitz** [comment]
That sounds like a good idea. 
This is not the current behavior that sends the whole document. 
I'll suggest that you can:
- Move the update process to a function.
- Create a feature request [here](https://github.com/appwrite/appwrite/issues)

**Comment by stnguyen90** [comment, marked as answer]
I would suggest maintaining a duplicate collection that contains the old data. When an update occurs, you can check the duplicate collection for the old value, do what you need, and then update the document in the duplicate collection.

Also, I would recommend 👍 this issue: https://github.com/appwrite/appwrite/issues/3294
